smartsim/settings/base.py

```diff
@@ -357,10 +357,25 @@ def run_command(self):
     def update_env(self, env_vars):
         """Update the job environment variables
 
+        .. note:
+
+            To fully inherit the current user environment, add the
+            workload-manager-specific flag to the launch command through the
+            :meth:`add_exe_args` method. For example, ``--export=ALL`` for
+            slurm, or ``-V`` for PBS/aprun.
+
+
         :param env_vars: environment variables to update or add
-        :type env_vars: dict[str, str]
+        :type env_vars: dict[str, Union[str, int, float, bool]]
+        :raises TypeError: if env_vars values cannot be coerced to strings
         """
-        self.env_vars.update(env_vars)
+        val_types = (str, int, float, bool)
+        # Coerce env_vars values to str as a convenience to user
+        for (env, val) in env_vars.items():
+            if type(val) not in val_types:
+                raise TypeError(f"env_vars[{env}] was of type {type(val)}, not {val_types_union}")
+            else:
+                self.env_vars[env] = val
```

Comment:
Shouldn't this be `self.env_vars[env] = str(val)`? Or, said differently, how are variables coerced to strings?

Reply:
OK, I see, we do it later, that part of the code was collapsed.

Reply:
Yeah, we could potentially cast types here, but I ended up leaving the behavior as is for this PR since we cast elsewhere and tests already exist to lock in the current behavior of not casting.